content/Proposals/Done/202106-proposals-process.md

```diff
@@ -0,0 +1,112 @@
+* Existing Google doc-based documents are hard to consume. The formatting is widely different. Naming is inconsistent.
+* Document creation is rarely actionable. There is no review process, so the effort of creating a relevant document might be wasted, as the document is lost. This also leads to docs being in the half-completed state, demotivating readers to look at it.
+* It's hard to track previous discussions around proposals, who approved them (e.g. proposals).
+* It's not public, and it's hard to share good proposals with other external and internal teams.
+
+## Goals
+
+Goals and use cases for the solution as proposed in [How](#how):
+
+* Allow easy collaboration and decision making on design ideas.
+* Have a consistent design style that is readable and understandable.
+* Ensure design docs are discoverable for better awareness and knowledge sharing about past decisions.
+* Define a clear review and approval process.
+
+## Non-Goals
+
+* Define process for other documents (see [handbook proposal](../Accepted/202106-handbook.md#pitfalls-of-the-current-solution))
+
+## How
+
+We want to propose an improved, official proposal process for Monitoring Group that clearly states *when, where and how* to create proposal/enhancement/design documents.
+
+Everything starts with a problem statement. It might be a missing functionality, confusing existing functionality or broken one. It might be an annoying process, performance or security issue (or potential one).
+
+### What Process to Follow?
+
+As defined in [handbook proposal](../Accepted/202106-handbook.md#pitfalls-of-the-current-solution), our Handbook should tell you that Handbook is meant to be an index for our team resources and linking point to other distributed projects we maintain or contribute to.
+
+First, we need to identify if the idea we have is something we can contribute to the upstream project, or it does not fit anywhere else, so we can leverage the [Handbok Proposal directory](..) and the [process](#handbook-proposal-process). See the below algorithm to find it out:
+
+![where](../../assets/proposal-where.png)
+
+[Internal Team Drive for Public and Confidential Proposals](https://drive.google.com/drive/folders/1WGqC3gMCxIQlrnjDUYfNUTPYYRI5Cxto)
+
+### Handbook Proposal Process
+
+If there is no problem, there is no need for changing anything, no need for a proposal. This might feel trivial, but we should first ask ourselves this question before even thinking about writing a proposal.
+
+It takes time to propose an idea, find consensus and implement more significant concepts, so let's not waste time before it's worth it. But, unfortunately, even good ideas sometimes have to wait for a good moment to discuss them.
+
+Let's assume the idea sounds interesting to you; what to do next, where to propose it? How to review it? Follow the algorithm below:
+
+![where](../../assets/proposal-how.png)
```

Comment:
I am a bit concerned with having only leads and architects as reviewers. That amounts to a pool of 3 people out of ~20 who can add feedback and approve changes. It also inhibits knowledge sharing and collaboration within the team since other members will be left out.

Reply:
I agree, was thinking about this too. Everyone should review. But question is, who is making final decision? What is the final vote? I would be totally for equal vote priority, but this would mean you have to have 11 out of 20 ppl to review which is unrealistic too. So at then end lead and architects has to know about it no?

This is BTW rare: What it will come next is that I will propose same to CMO, Observatorium, Prometheus Operator, Prometheus, Thanos etc, and then normal procedure applies as to any bigger change (consensus in maintainers team)

Reply:
Any proposed flow? Just get it reviewed by interested team members you choose? Consensus in the team?

Reply:
I think it makes complete sense to have the architect and lead from the team responsible for the topic as reviewers. 

I would, however, also encourage involving team members with the subject-specific domain knowledge in the review process. We have a list of contributors in [this doc](https://docs.google.com/spreadsheets/d/1plCB4Q-IB16_LhFiZiiZ3F9LPT-lHraphcm8D8Wusd8/edit#gid=1467586438) that we can use to select the appropriate team members.

In addition to that, anyone should feel free to comment on the proposal, regardless of whether they're selected as reviewers or not. In my view, a proposal should be treated as a tool for collaboration, innovation and dissipating knowledge within the team.

Reply:
Review should be open to everyone, even outside of the team: this is open-source after all and anyone can comment, which is free input for us.

We _do_ want some accountability when it comes to approval to ensure that things do not hang around in limbo indefinitely and ensure that proposals are either accepted or rejected. In a regular team sync a few weeks ago I originally recommended that we follow a structured process like this inspired by Hashicorp's RFC processe [0]:

> Once you’ve written the first draft of an RFC, share it with your team. They’re likely to have the most context on your proposal and its potential impacts, so most of your feedback will probably come at this stage. Any team member can comment on and approve an RFC, but you need explicit approval only from the appropriate team leads in order to move forward. Once the RFC is approved and shared with stakeholders, you can start implementing the solution. For major projects, also share the RFC to the company-wide email list. While most members of the mailing list will just read the email rather than the full RFC, sending it to the list gives visibility into major decisions being made across the company. 

In our case, the architect would be ultimately responsible for signing off on something, but ANYONE can give feedback. We can even say: X people are chosen to review and give expert input on every document and present recommendations to the architect who ultimately signs off (or not). Alternatively, these X people could approve/reject the document by consensus.

[0]

Reply:
Added mention to that to our process